cmd/secrets-provider/main.go

```diff
@@ -51,63 +51,51 @@ func main() {
 		printErrorAndExit(messages.CSPFK009E)
 	}
 
-	// Configure exponential backoff
-	expBackoff := backoff.NewExponentialBackOff()
-	expBackoff.InitialInterval = 2 * time.Second
-	expBackoff.RandomizationFactor = 0.5
-	expBackoff.Multiplier = 2
-	expBackoff.MaxInterval = 15 * time.Second
-	expBackoff.MaxElapsedTime = 2 * time.Minute
-
-	err = backoff.Retry(func() error {
-		for {
-			log.Info(fmt.Sprintf(messages.CSPFK001I, authn.Config.Username))
-			authnResp, err := authn.Authenticate()
-			if err != nil {
-				return log.RecordedError(messages.CSPFK010E)
-			}
-
-			err = authn.ParseAuthenticationResponse(authnResp)
-			if err != nil {
-				return log.RecordedError(messages.CSPFK011E)
-			}
-
-			err = provideConjurSecrets(accessToken)
-			if err != nil {
-				return log.RecordedError(messages.CSPFK016E)
-			}
-
+	constantBackOff := backoff.NewConstantBackOff(time.Duration(secretsConfig.SyncIntervalTime) * time.Second)
+	count := 0
+	backoff.Retry(func() error {
+		if count > secretsConfig.RetryCount {
+			// Deleting the retrieved Conjur access token in case we got an error after retrieval.
+			// if the access token is already deleted the action should not fail
 			err = accessToken.Delete()
 			if err != nil {
-				return log.RecordedError(messages.CSPFK003E, err.Error())
-			}
-
-			if authnConfig.ContainerMode == "init" || authnConfig.ContainerMode == "application" {
-				log.Info(messages.CSPFK010I)
-				os.Exit(0)
+				log.Error(messages.CSPFK003E, err)
 			}
+			log.Error(messages.CSPFK038E)
+			printErrorAndExit(messages.CSPFK039E)
+		}
+		if count > 0 {
+			log.Info(fmt.Sprintf(messages.CSPFK010I, count, secretsConfig.RetryCount))
+		}
+		count++
 
-			// Reset exponential backoff
-			expBackoff.Reset()
+		log.Info(fmt.Sprintf(messages.CSPFK001I, authn.Config.Username))
+		authnResp, err := authn.Authenticate()
+		if err != nil {
+			return log.RecordedError(messages.CSPFK010E)
```

Comment:
i may be missing something but it seems that you are returning this error but it is never used. am i wrong?

Reply:
Returning this error triggers the retry loop.